DJAN-107 tequila adoption

jbradberry opened the pull request:
Merging together the old `common` and `structure` roles, and splitting off into its own repo, to support installation using `ansible-galaxy`.

jbradberry committed:
Moved the creation of the project database and database user to db-master.

These are tasks that are tied to the database service, regardless of
whether the db is local or on another server.

jbradberry committed:
Improved the installation of rabbitmq.

- add the erlang ppa when the Ubuntu release is 'precise'
- get the latest version of rabbitmq and erlang-nox

jbradberry committed:
Merged the cache role into web.

jbradberry committed:
Merged the web role into the balancer role.

jbradberry committed:
Merged structure role into common.

jbradberry committed:
Removed the empty worker role.

jbradberry committed:
Removed the remainder of the role metadata.

jbradberry committed:
Lifted the common role out to the top level.

And removed all other roles and unnecessary files.

jbradberry committed:
Added ansible-galaxy metadata.

jbradberry committed:
Added tasks to install and set up newrelic system monitoring.

jbradberry committed:
Made the newrelic tasks conditional on the license key being defined.

jbradberry committed:
Cleaned up the README to information relevant to the role.

* removed the project structure information
* specified how to install the role using ansible-galaxy
* listed all of the variables actually used by this role

jbradberry committed:
Moved out the variable defaults that are not relevant for this role.

dpoirier commented:
Here and elsewhere, I'm wondering when we have more than a couple tasks in a row that only run when the same condition is true, if it'd be clearer if we moved those tasks to another file and used a conditional include instead of having them all in main?

dpoirier commented on tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+    - restart fail2ban
+
+- name: add github to the ssh known hosts
+  lineinfile: line="{{ lookup('file', 'github.pub') }}"
```
we can use the `known_hosts` module here (Ansible 1.9+)

jbradberry commented:
Hm, intriguing.  Yes, I think I might like that.  (The conditional includes, that is.)

dpoirier commented:
Or we could use the new `block` feature in Ansible 2.0 if we're willing to require Ansible 2.0.  (Which wouldn't bother me at all.)

jbradberry committed:
Split out the optional newrelic tasks into their own file.

jbradberry commented:
I went with the conditional include, for now.

jbradberry committed:
Switched to using the known_hosts module for the github key.

jbradberry committed:
Minor tweaks.

* removed the no longer needed .gitignore file
* updated the year in LICENSE

dpoirier commented:
We need to do an "apt-get update" before trying to install any packages.

dpoirier commented:
Can we add something like this as the first task, to prevent a bad configuration from locking everyone out of the deployed systems?

```
- name: make sure we've defined some dev users
  fail: msg="Please define one or more developers in the `users` variable"
  when: users|length == 0
```

jbradberry committed:
Additional adjustments

- make `users` be a dict of dicts, like in Margarita
- fail out if `users` is not defined, so we can't get locked out
- update the apt cache if it hasn't been updated in the last hour

jbradberry commented:
@poirier r?  I believe I've addressed all of your review comments.

jbradberry committed:
Make New Relic conditional on a `use_newrelic` boolean variable

For consistency with a recommendation on tequila-django.

dpoirier commented:
:shipit:  I haven't tried it, but I assume you have and it looks reasonable.

dpoirier commented:
P.S. at work I'm @dpoirier.   The poirier ID is my personal account.

jbradberry committed:
Update developer ssh key management to do a full nested iteration

so we get the full setup of each user's full set of keys.

jbradberry committed:
Final tweaks
